Update MTENN to include MSE step loss from `drugforge`

khuddzu opened the pull request:
Adding MSELoss from drugforge, to include step loss functionality. If we are okay with adding drugforge dependency, then I will add using their package. If not, I can hardcode their loss here, which is what I will start with anyways. There are other loss functions as well they have implemented that we could access.

Will close #348 

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenADMET/openadmet_toolkit/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

khuddzu committed:
Hardcode steploss into mtenn

khuddzu committed:
Update mtenn to include step loss functionality; dirty

hmacdope commented:
@khuddzu hardcoding the loss is preferable, as `drugforge` has an insane amount of dependencies.

hmacdope reviewed (changes requested):
Might be worth abstracting this into a `loss.py` setup with some of our custom loss functions.

khuddzu committed:
Merge branch 'feat/369/mtenn_predict_compat' into feat/348/mtenn_loss
Merge to create single workspace for mtenn debugging

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#374)

updates:
- https://github.com/psf/black → https://github.com/psf/black-pre-commit-mirror
- [github.com/PyCQA/isort: 6.0.1 → 6.1.0](https://github.com/PyCQA/isort/compare/6.0.1...6.1.0)
- [github.com/astral-sh/ruff-pre-commit: v0.13.2 → v0.13.3](https://github.com/astral-sh/ruff-pre-commit/compare/v0.13.2...v0.13.3)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

khuddzu committed:
Final sweep of errors I have found in API and docs (#375)

* Change docstring to reflect pytorch models not sklearn.

* Update docstring for DescriptorFeaturizer to reflect descriptors, not fingerprints

* Update link error, comment, and docstring in anvil ref

* Change OpenFE to OpenADMET

cynthiaxu04 committed:
Add x error bars to plotting (#379)

* Add x error bars to plotting

* Update regplot docstring with new argument

* Correct based on PR feedback

dwwest committed:
Bugfixes and updates to posthoc comparison to streamline user experience (#300)

* Add model-dir to replace model-stats file

* Add label and taskname from anvil function

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add function to read in anvil recipes and cross validation metrics from a parent folder

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add chemeleon special case for model labeling

* Allows posthoc to now cleanly fail if multiple identical labels are provided for comparison

* Fix error with path vs Path for click in compare cli

* Fix spelling mistake in compare cli

* Update tests with new arg names

* Add better unit testing for posthoc

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add documentation and default types to functions in comparison

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add featurizer label to be read from anvil plus testing

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix color issues, fix cli to add new arguments, clean up docstrings, rename arguments to be consistent between cli and posthoc code

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix comment issues after merge

* Add preliminary function for reading from s3 buckets

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add boto3 to the conda install

* Add integration test for comparison

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix argument name in posthoc unit tests

* Fix argument names in posthoc tests again

* Fix issue in posthoc anvil integration test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix anvil test tmp_folder issue

* Fix posthoc integration test argument calls

* Fix posthoc testing issues with incorrect filepaths

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add comments for posthoc testing

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add Sean's fixes

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
CLI Guide (#358)

* add cli guide files

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update posthoc cli based on pending changes from PR

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* apply suggestions from @khuddzu

* Kate suggestion

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Devany West <devanywest314@gmail.com>

khuddzu committed:
Fix indentation error in featurizer

smcolby committed:
Set shuffle default in GAT featurizer to `False` (#385)

* Sort imports

* Set default shuffle behavior to False

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#390)

updates:
- [github.com/PyCQA/isort: 6.1.0 → 7.0.0](https://github.com/PyCQA/isort/compare/6.1.0...7.0.0)
- [github.com/asottile/pyupgrade: v3.20.0 → v3.21.0](https://github.com/asottile/pyupgrade/compare/v3.20.0...v3.21.0)
- [github.com/astral-sh/ruff-pre-commit: v0.13.3 → v0.14.0](https://github.com/astral-sh/ruff-pre-commit/compare/v0.13.3...v0.14.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

khuddzu committed:
Create a warning if MTENN is being used with CPU (#386)

* Add gpu dependency to predict

* Change error message to warning

* Make warning more obnoxious

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add line to satisfy linter

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add line to satisfy linter

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* try new approach to satify linter

* Add line after removal (even though all other functions dont have a blank line) to satisfy linter

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add blank line before warning

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update warning to env variable

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add new variable to predict docstring

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
Add possibility to have no calibration in `Committie Regressor` via Anvil (#388)

* add possibility to pass through no calibration

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add ability to fail on cal model load

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* try fix docstring

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update openadmet/models/anvil/specification.py

Co-authored-by: Sean M. Colby <seanmcolby@gmail.com>

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Sean M. Colby <seanmcolby@gmail.com>

hmacdope committed:
Try and trim test runtimes and add duration monitoring (#377)

* try and trim runtimes and add duration monitoring

* truncate train data

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* try and fix fit-transform issues

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* remove redundant anvil runs

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
add warning to GAT (#392)

hmacdope committed:
Change CV to work on whole dataset rather than splits (#394)

* start modification of trad-ml CV

* change data input to CV

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* docstrings

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dwwest committed:
Add nepare model to architectures (#234)

* Add nepare file to architecture

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add first non-functional wrapper for nepare

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Reorgs some code for nepare, non functional

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Adds skeleton of pairwise featurizer, needs work

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Adds pairwise featurizer class and tests

* Refactors pairing data to new class in anvil workflow 'augmentation'

* Removes pairing from main anvil workflow

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Reorgs some files out of augment

* Add PairedFeaturizer that can take other featurizers, featurize, and then load into nepare dataloader

* Pairwise fix

* Working version of NepareChempropFeaturizer, no scaler

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Remove augment import from anvil

* Add nepare-chemprop architecture

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add nepare feat registration

* Commit rest of changes before refactor

* Rework nepare architecture to more generic paired featurizer

* Rewrite Nepare module within openadmet framework

* Fix monitor metric issue

* Fix eval step val_loss naming issue

* Add t_true and t_pred updates for pairwise data

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add logging statements to CV, and address field validator issues in pairwise featuirizer

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add pairwise featurizer test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix regression issue

* Fix make new function in pairwise featurizer

* Fix cross validation indexing error

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add docstrings for nepare

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add docstrings to top of nepare files

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add attempt to include input_dim in anvil workflow

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix input_dim bug in workflow

* Fix issue with input_dim not a param for most builds

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix import error in nepare unit test

* Fix issue in nepare test anvil file and fix train_ensemble input_dim issue

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Correct inference issue, now will error if parquet or other file not csv or sdf is passed

* Add fixes so nepare models can run in inference mode

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add changes from Hugo's review

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix kwargs issue in workflow for input_dim

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix issue in prediction step and t_pred t_true

* Add changes for Sean's review

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix hash issue with nepare module and introduce sampling for evaluation metrics to improve runtime

* Add back failing chemprop_mt recipe to integration test

* Fix for mask nans in eval

* Add Sean's review

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add report

* Add report section back, still with error

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix pydantic issue in pairwise featurizer

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Remove duplicate report in test data

* Remove duplicate regression metrics

* Fix randomizer for nepare

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add Sean's review comments

* Fix field validator issue for pydantic

* Fix dict issue in how_to_pair

* Add class method to pairwise

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: hmacdope <hugomacdermott@gmail.com>

hmacdope committed:
Add check to error on ensemble and CV (#396)

* add option to error on ensemble and CV

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
Add ability to have predefined splits. (#395)

* start modification of trad-ml CV

* change data input to CV

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* docstrings

* use WIP

* semi-working

* fix issues with resource not being present

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add testing setup

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add docs

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

khuddzu committed:
Add different hierarchy for headings (#399)

* Add different hierarchy for headings

* Apply suggestions from code review

Co-authored-by: Sean M. Colby <seanmcolby@gmail.com>

---------

Co-authored-by: Hugo MacDermott-Opeskin <hugomacdermott@gmail.com>
Co-authored-by: Sean M. Colby <seanmcolby@gmail.com>

hmacdope committed:
Change link to OpenADMET Models demo (#400)

Updated the link to OpenADMET Models showcase.

hmacdope committed:
Update project title in README.md

smcolby committed:
Load in best weights after training (#401)

* Load in best weights after training

* Load weights in a more consistent way

* Set `weights_only` to True

* Set `weights_only` to False

hmacdope committed:
Add guardrail for multitasking  (#378)

* add basic guardrail

* simplify

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* return

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fix merge conf

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* change to private attr

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* add model validator

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dwwest committed:
Add handling for nans from dummy models for #276 (#402)

* Add handling for nans from dummy models

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
Add verbose parameter to LGBM model (#405)

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#404)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.14.0 → v0.14.1](https://github.com/astral-sh/ruff-pre-commit/compare/v0.14.0...v0.14.1)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

cynthiaxu04 committed:
Anchor plots to 0 (#397)

* Anchor plots to 0

* Anchor posthoc anova plots to 0

cynthiaxu04 committed:
Add link to demos.openadmet.org on docs main page (#407)

* Add link to demos.openadmet.org on docs main page

* Update docs/index.rst

Co-authored-by: Devany West <31937638+dwwest@users.noreply.github.com>

---------

Co-authored-by: Devany West <31937638+dwwest@users.noreply.github.com>

hmacdope committed:
Update README with new links and remove examples

Removed outdated examples and notes about repo stability. Updated documentation links and added badges for demos and Google Colab.

hmacdope committed:
Cross link demos into documentation (#409)

* XLink to demos

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* bump CI for timing

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hmacdope committed:
Replace Google Colab link with Binder link in README

Updated the README to replace Google Colab link with Binder link for live showcase examples.

hmacdope committed:
small docs fixes

hmacdope committed:
Update CI badge link in README.md

hmacdope committed:
Merge remote-tracking branch 'origin/main' into feat/348/mtenn_loss

khuddzu committed:
Merge branch 'main' into feat/348/mtenn_loss

khuddzu committed:
Ensure mask_nans sees 1D arrays

khuddzu committed:
Add validation step to mtenn
